Tokeninfo "client id or secret is null"

### General Information

- Mode (Sandbox/Live): Live
- PayPal-Debug-ID(s) (from any logs): 3c65a55c453d4
- Version of Java used: 1.8.0_77
- SDK Version: 1.4.2

### Issue Description

I've got problem with method Tokeninfo.createFromAuthorizationCode.
In Sandbox mode everything is OK, but after switching to Live mode I receive
````
com.paypal.base.rest.PayPalRESTException: Error code : 400 with response : {"error_description":"client id or secret is null","error":"invalid_client","correlation_id":"3c65a55c453d4","information_link":"https://developer.paypal.com/docs/api/#errors"}
	at com.paypal.base.rest.PayPalRESTException.createFromHttpErrorException(PayPalRESTException.java:72)
	at com.paypal.base.rest.PayPalResource.execute(PayPalResource.java:439)
	at com.paypal.base.rest.PayPalResource.configureAndExecute(PayPalResource.java:267)
	at com.paypal.api.openidconnect.Tokeninfo.createFromAuthorizationCodeParameters(Tokeninfo.java:254)
	at com.paypal.api.openidconnect.Tokeninfo.createFromAuthorizationCode(Tokeninfo.java:194)
	at controllers.saas.PaymentController.returnUrl(PaymentController.java:189)
	at router.Routes$$anonfun$routes$1$$anonfun$applyOrElse$80$$anonfun$apply$80.apply(Routes.scala:2098)
	at router.Routes$$anonfun$routes$1$$anonfun$applyOrElse$80$$anonfun$apply$80.apply(Routes.scala:2098)
	at play.core.routing.HandlerInvokerFactory$$anon$4.resultCall(HandlerInvoker.scala:157)
	at play.core.routing.HandlerInvokerFactory$$anon$4.resultCall(HandlerInvoker.scala:156)
Caused by: com.paypal.base.exception.HttpErrorException: Error code : 400 with response : {"error_description":"client id or secret is null","error":"invalid_client","correlation_id":"3c65a55c453d4","information_link":"https://developer.paypal.com/docs/api/#errors"}
	at com.paypal.base.HttpConnection.executeWithStream(HttpConnection.java:174)
	at com.paypal.base.HttpConnection.execute(HttpConnection.java:67)
	at com.paypal.base.rest.PayPalResource.execute(PayPalResource.java:425)
	at com.paypal.base.rest.PayPalResource.configureAndExecute(PayPalResource.java:267)
	at com.paypal.api.openidconnect.Tokeninfo.createFromAuthorizationCodeParameters(Tokeninfo.java:254)
	at com.paypal.api.openidconnect.Tokeninfo.createFromAuthorizationCode(Tokeninfo.java:194)
	at controllers.saas.PaymentController.returnUrl(PaymentController.java:189)
	at router.Routes$$anonfun$routes$1$$anonfun$applyOrElse$80$$anonfun$apply$80.apply(Routes.scala:2098)
	at router.Routes$$anonfun$routes$1$$anonfun$applyOrElse$80$$anonfun$apply$80.apply(Routes.scala:2098)
	at play.core.routing.HandlerInvokerFactory$$anon$4.resultCall(HandlerInvoker.scala:157)
Caused by: java.io.IOException: Server returned HTTP response code: 400 for URL: https://api.paypal.com/v1/identity/openidconnect/tokenservice
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at sun.net.www.protocol.http.HttpURLConnection$10.run(HttpURLConnection.java:1890)
	at sun.net.www.protocol.http.HttpURLConnection$10.run(HttpURLConnection.java:1885)
	at java.security.AccessController.doPrivileged(Native Method)
	at sun.net.www.protocol.http.HttpURLConnection.getChainedException(HttpURLConnection.java:1884)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1457)
	at sun.net.www.protocol.http.HttpURLConnection.access$200(HttpURLConnection.java:90)
Caused by: java.io.IOException: Server returned HTTP response code: 400 for URL: https://api.paypal.com/v1/identity/openidconnect/tokenservice
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1840)
	at sun.net.www.protocol.http.HttpURLConnection.access$200(HttpURLConnection.java:90)
	at sun.net.www.protocol.http.HttpURLConnection$9.run(HttpURLConnection.java:1433)
	at sun.net.www.protocol.http.HttpURLConnection$9.run(HttpURLConnection.java:1431)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.AccessController.doPrivilegedWithCombiner(AccessController.java:782)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1430)
	at java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:480)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:338)
	at com.paypal.base.HttpConnection.executeWithStream(HttpConnection.java:133)
```


I think problem is in class HttpConnection. Method setHttpHeaders is called only in sandbox mode.

--- a/rest-api-sdk/src/main/java/com/paypal/base/HttpConnection.java
+++ b/rest-api-sdk/src/main/java/com/paypal/base/HttpConnection.java
@@ -101,12 +101,12 @@ public abstract class HttpConnection {
 		connection.setRequestProperty("Content-Length", ""
 				+ payload.trim().length());
 		try {
+			setHttpHeaders(headers);
 			String mode = ConfigManager.getInstance().getConfigurationMap()
 					.get(Constants.MODE);
 			if (headers != null && !Constants.LIVE.equalsIgnoreCase(mode)) {
 				String cmd = "curl command: \n";
 				cmd += "curl -v '" + connection.getURL().toString() + "' \\\n";
-				setHttpHeaders(headers);
 				Iterator<String> keyIter = headers.keySet().iterator();
 				while (keyIter.hasNext()) {
 					String key = keyIter.next();
@@ -252,12 +252,14 @@ public abstract class HttpConnection {
 	 * @param headers
 	 */
 	protected void setHttpHeaders(Map<String, String> headers) {
-		Iterator<Map.Entry<String, String>> itr = headers.entrySet().iterator();
-		while (itr.hasNext()) {
-			Map.Entry<String, String> pairs = itr.next();
-			String key = pairs.getKey();
-			String value = pairs.getValue();
-			this.connection.setRequestProperty(key, value);
+		if (headers != null && !headers.isEmpty()) {
+			Iterator<Map.Entry<String, String>> itr = headers.entrySet().iterator();
+			while (itr.hasNext()) {
+				Map.Entry<String, String> pairs = itr.next();
+				String key = pairs.getKey();
+				String value = pairs.getValue();
+				this.connection.setRequestProperty(key, value);
+			}
 		}
 	}
 

